File: RepoTransBench/target_projects/Java/C#/AllenDowney_ThinkJavaCode/public_tests/Ch03/ConvertPublicTest.cs
using Xunit;
using AllenDowney.ThinkJavaCode;

namespace PublicTests.Ch03
{
    public class ConvertPublicTest
    {
        [Fact]
        public void TestCelsiusToFahrenheitPublicData()
        {
            Assert.Equal(32.0, Convert.CelsiusToFahrenheit(0), 2);
            Assert.Equal(122.0, Convert.CelsiusToFahrenheit(50), 2);
            Assert.Equal(68.0, Convert.CelsiusToFahrenheit(20), 2);
            Assert.Equal(98.6, Convert.CelsiusToFahrenheit(37), 2);
        }

        [Fact]
        public void TestFahrenheitToCelsiusPublicData()
        {
            Assert.Equal(0.0, Convert.FahrenheitToCelsius(32), 2);
            Assert.Equal(100.0, Convert.FahrenheitToCelsius(212), 2);
            Assert.Equal(40.0, Convert.FahrenheitToCelsius(104), 2);
            Assert.Equal(5.0, Convert.FahrenheitToCelsius(41), 2);
        }
    }
}

File: RepoTransBench/source_projects/Java/AllenDowney_ThinkJavaCode/src/test/java/ch03/ConvertPublicTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class ConvertPublicTest {

    @Test
    public void testCelsiusToFahrenheitPublicData() {
        // 0C = 32F, 50C = 122F, 20C = 68F, 37C = 98.6F
        assertEquals(32.0, ch03.Convert.celsiusToFahrenheit(0), 0.01);
        assertEquals(122.0, ch03.Convert.celsiusToFahrenheit(50), 0.01);
        assertEquals(68.0, ch03.Convert.celsiusToFahrenheit(20), 0.01);
        assertEquals(98.6, ch03.Convert.celsiusToFahrenheit(37), 0.01);
    }

    @Test
    public void testFahrenheitToCelsiusPublicData() {
        // 32F=0C, 212F=100C, 104F=40C, 41F=5C
        assertEquals(0.0, ch03.Convert.fahrenheitToCelsius(32), 0.01);
        assertEquals(100.0, ch03.Convert.fahrenheitToCelsius(212), 0.01);
        assertEquals(40.0, ch03.Convert.fahrenheitToCelsius(104), 0.01);
        assertEquals(5.0, ch03.Convert.fahrenheitToCelsius(41), 0.01);
    }
}